### ISSUE TITLE
Source map per URL host

### ISSUE THREAD

**Issue by kobiburnley** [SponsorPay/stack-retrace#9]
Stack-traces may contain files from different vendors, so their corresponding sourcemaps are not in the same file system location/http server. Would we want to implement provider-per-URL-host mechanism?

@mckomo

**Comment by mckomo** [comment]
Yes, of course. I was thinking that we could implement `SourceMapProvider` in the "router" style that could compose core source map providers. It could look like this:

```js
const sourceMapProvider = routedProvider({
  "cdn.com/*": httpProvider,
  "*": fsProvider("/var/sources")
})
```
